woocommerce/woocommerce-android
# Add WireMock network mocking for screenshot test

Adds infrastructure for network mocking in connected tests with WireMock.
Also adds an initial set of mocks for the screenshot test.
(Internal ref: paaHJt-Pp-p2)

### Changes

* This uses [WireMock](http://wiremock.org/) to power a mock server for the network requests.
* WireMock runs a small HTTP server on `localhost:8080` which mimics the WordPress.com API. Request/response configuration is done with the JSON files (loaded as assets).
* The implementation is based on the WordPress Android network mocking setup. (In future we may want to share this implementation more directly.)
* The network security config is updated to allow clear text traffic, to support WireMock running the mock server on `localhost` over HTTP. We could look at making changes to FluxC to support another configuration, but I wasn't able to do that within the scope of this work.

### To test

1. Ensure your `gradle.properties` includes `wc.screenshots.url`, `wc.screenshots.username`, and `wc.screenshots.password`. (These can be set to the same example values as in `gradle-properties.example`.)
2. Run the connected test in `ScreenshotTest`.

### To review

I'd appreciate at least one review from the Platform9 team and one review from a Woo Android developer, looking at both the infrastructure and changes to the app itself.

Note: Although the diff for this PR is big, it's mostly due to the JSON files (mocked responses). You can view the changes without those JSON files [here](https://github.com/woocommerce/woocommerce-android/pull/2933/files?file-filters%5B%5D=.gradle&file-filters%5B%5D=.java&file-filters%5B%5D=.kt&file-filters%5B%5D=.xml).

### Update release notes:
